flink-core/src/main/java/org/apache/flink/api/connector/source/lib/DataGeneratorSource.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.connector.source.lib;
```

Comment:
How did you arrive at that package name? It's rather non-descriptive.

Reply:
Great question. This is the layout that Stephan chose for the `NumberSequenceSource`, which is very close in nature to the `DataGeneratorSource`, so I did not dwell too much on that decision.

Reply:
How about such layout? The NumberSequenceSource should ideally also be in the datagen package, but it is already @Public.
![image](https://user-images.githubusercontent.com/1492164/189342430-a622f9e4-6d11-4d31-b777-a0916ccad69b.png)

Reply:
I think that would be better. Moving the remaining classes is an orthogonal thing we can take separately. (which'd be a 2 step-process, first moving the classes and leaving behind a sort of proxy `Source extends org.apache.flink...Source` that is deprecated), that we then remove whenever we see fit.

Reply:
Applied the changes as discussed.